test/erc-20-extensions/ERC20Extensions.js

```diff
@@ -0,0 +1,203 @@
+    });
+
+    it("should revert when trying to pause the contract when it's already paused", async function () {
+      await ERC20Pausable.pause();
+      await expect(ERC20Pausable.pause()).to.be.reverted;
+    });
+
+    it("should revert when trying to mint tokens while paused", async function () {
+      await ERC20Pausable.pause();
+      await expect(ERC20Pausable.mint(addr1.address, amount)).to.be.reverted;
+    });
+
+    it("should revert when a non-owner tries to pause or unpause the contract", async function () {
+      // Expect pause to be reverted when called by a non-owner
+      await expect(ERC20Pausable.connect(addr1).pause()).to.be.reverted;
+
+      // Pause the contract
+      await ERC20Pausable.pause();
+
+      // Expect unpause to be reverted when called by a non-owner
+      await expect(ERC20Pausable.connect(addr1).unpause()).to.be.reverted;
+    });
+  });
+
+  describe("ERC20Snapshot tests", function () {
+    it("should create a new snapshot and emit a Snapshot event", async function () {
+      // Create a new snapshot and wait for the transaction receipt
+      const tx = await ERC20Snapshot.snapshot();
+      const receipt = await tx.wait();
+
+      // Verify that "Snapshot" event is emited 
+      expect(receipt.events[0].event).to.equal("Snapshot");
+    });
+
+    it("should return the correct totalSupplyAt(snapshotId)", async function () {
+      // Create a new snapshot and wait for the transaction receipt
+      const snapshot = await ERC20Snapshot.snapshot();
+      const tx = await snapshot.wait();
+
+      // Get the snapshotID
+      const snapshotId = tx.events[0].args[0];
+
+      // Mint extra tokens, increase the total supply and get the current total supply
+      await ERC20Snapshot.mint(owner.address, amount);
+      const newTotalSupply = await ERC20Snapshot.totalSupply();
+
+      //Verify that the total supply at the time of the snapshot is equal to the initial mint amount.
+      expect(await ERC20Snapshot.totalSupplyAt(snapshotId)).to.equal(newTotalSupply - amount);
+    });
+
+    it("should return the correct balanceOfAt(address, snapshotId)", async function () {
+      await ERC20Snapshot.transfer(addr1.address, transferAmount);
+
+      // Create a new snapshot and wait for the transaction receipt
+      const snapshot = await ERC20Snapshot.snapshot();
+      const tx = await snapshot.wait();
+
+      // Get the snapshotID
+      const snapshotId = tx.events[0].args[0];
+
+      // Transfer extra tokens to addr1 to verify that the snapshot is valid and get new ballance
+      await ERC20Snapshot.transfer(addr1.address, transferAmount);
+      const newBalance = await ERC20Snapshot.balanceOf(addr1.address)
+
+      //Verify that the balance of addr1 at the time of the snapshot is equal to the initial transfer amount.
+      expect(await ERC20Snapshot.balanceOfAt(addr1.address, snapshotId)).to.equal(newBalance - transferAmount);
+    });
+  });
+});
```

Comment:
nit: new lines at the end of this and the `.sol` file.

Reply:
done